### DISCUSSION TITLE
Some ideas to make project better!

### DISCUSSION THREAD

**Discussion by DarkXero-dev** [prasanthrangan/hyprdots#634] [Ideas]
Hey there

As I was discussing over at issue **[#629](https://github.com/prasanthrangan/hyprdots/issues/629)** I would love to help make this a more viable option for other users. Here are the points I have... I agree that this should not be turned into a Distro, too complex to maintain, I should know lol. Post-Install script is best thing for the project ;)

- Have a Repo of pre-built packages for easier/faster install.
- Do not forget to add a License to the repo to gather more "Trust"
- Fix the issue where **nvidia** modules not being added to **mkinitcpio.conf**
- Blacklist **nouveau** drivers in **Grub** for recent **nVidia** cards...
- Maybe include **[XWaylandVideoBridge](https://aur.archlinux.org/packages/xwaylandvideobridge-bin)** for better screen/window sharing.
- Add **Flatpak** support out the box coz more n more users use them.
- Replace **Dunst** with the more customizable **[SwayNC](https://github.com/ErikReider/SwayNotificationCenter)**
- Replace **Kitty** with well cutomized **Konsole** since we using KDE elements.
- Include a **Grub** theme making it all more consistent.
- More **SDDM** themes that fit rest of styles.
- Find a way to make **Flatpaks** respect chosen theme like **[my thread](https://forum.xerolinux.xyz/thread-152.html)**
- Maybe implement **[nwg-displays](https://github.com/nwg-piotr/nwg-displays)** making it easier to config monitors
- Have a choice if user wants a dock or not with **[nwg-dock](https://github.com/nwg-piotr/nwg-dock)**
- Make script design a lot better (optional) making it more readable.. 
- Finally a well organized pkgs.lst like I did => **[Here](https://github.com/xerolinux/xero_iso/blob/main/XeroKDE/packages.x86_64)**

Here's Repo I created, let me know what you want me to add to it.
https://repos.xerolinux.xyz/xero_hypr/x86_64/

Can be added to pacman.conf

That's all I got for now.. I can help with repos should you accept collab.. With that I say best of luck and great job as usual ;)

**Comment by kRHYME7** [comment]
Hi first question, In any case, if one of the pre-compiled/built packages lags in their ``` pkgver ``` should the repo still follow the latest packages or the pre-build one?

**Reply by DarkXero-dev** [reply]
I will be taking care of packages, building them n hosting them, -git packages do not get updated coz they pull latest commits during build ..

**Reply by kRHYME7** [reply]
Wow! That's indeed beneficial. I'm curious if Chaotic AUR does the same thing? I am using Chaotic on my KDE times and before I  installed Hyprland,  Chaotic AUR was already on my system and if I remember correctly Installing hyprdots did not took much time as I think Chaotic hosts the compiled packages( But maybe Not all packages). 
I  did not dig deeper though.

**Reply by DarkXero-dev** [reply]
That too can be used they update daily. Great idea would save both of us the trouble.

**Reply by ghost** [reply]
Yeah the Chaotic AUR does host some packages needed for hyprdots, i've tried it before for this repo and it already decreases install time by like 20 minutes.

**Reply by DarkXero-dev** [reply]
Ok then, just let me know what packages do not exist on Chaotic AUR and I shall host them. Not a problem here. Anything to help ;)

**Reply by akashgagda** [reply]
> Ok then, just let me know what packages do not exist on Chaotic AUR and I shall host them. Not a problem here. Anything to help ;)

Thank you much bro for contributing

**Reply by kRHYME7** [reply]
How about we do some tagged released versions (with packages from @DarkXero-dev ). With these, users should always have a successful fresh installation. Then it's on them to update. @prasanthrangan 

I already added ``` Hyprots-install ``` a while back(WIP) so users only required to run this command. It will just run ``` install.sh ./custom_app.lst ```. I can modify this to be interactive (fzf) using the .lst files provided, if anyone is interested.

**Comment by Gigas002** [comment]
IMO would be awesome if we could at least turn on/off components on installation. Like, I didn't want to install a `vscode`, `kitty`, override my existing `grub` config (*thank god I read the scripts and made a backup, the installed one broken something for me*), etc, but the script will fail in the middle of process if something's missing, so I **had** to install all that stuff just to remove it later

**Reply by kRHYME7** [reply]
Now, we can restore a backup using [Hyprdots-ctl](https://github.com/kRHYME7/Hyprdots-ctl), but it would only restore the configs.

As for the system stuff, I guess a control file for that, But I don't know what might break, because this  repo is intended for a fresh install of Arch

In my case, I disabled grub and sddm flags and everything works fine. Maybe because I'm using systemd-boot and my grub is already broken.
